pyorbital/tlefile.py

```diff
@@ -366,49 +397,68 @@ def read_tle_files(self):
         paths = self.config["downloaders"]["read_tle_files"]["paths"]
 
         # Collect filenames
-        fnames = []
-        for path in paths:
-            if '*' in path:
-                fnames += glob.glob(path)
-            else:
-                if not os.path.exists(path):
-                    logging.error("File %s doesn't exist.", path)
-                    continue
-                fnames += [path]
+        fnames = collect_fnames(paths)
+        tles = _parse_tles_for_downloader(fnames, open)
+        logging.info("Loaded %d TLEs from local files", len(tles))
 
-        tles = []
-        for fname in fnames:
-            with open(fname, 'r') as fid:
-                data = fid.read()
-            tles += self.parse_tles(data)
+        return tles
 
-        logging.info("Loaded %d TLEs from local files", len(tles))
+    def read_xml_admin_messages(self):
+        """Read Eumetsat admin messages in XML format."""
+        paths = self.config["downloaders"]["read_xml_admin_messages"]["paths"]
+        tles = read_tles_from_mmam_xml_files(paths)
+        logging.info("Loaded %d TLEs from admin message XML files", len(tles))
 
         return tles
 
-    def parse_tles(self, raw_data):
-        """Parse all the TLEs in the given raw text data."""
-        tles = []
-        line1, line2 = None, None
-        raw_data = raw_data.split('\n')
-        for row in raw_data:
-            if row.startswith('1 '):
-                line1 = row
-            elif row.startswith('2 '):
-                line2 = row
-            else:
+
+def _parse_tles_for_downloader(item, open_func):
+    return [Tle('', tle_file=io.StringIO(tle)) for tle in
+            _get_tles_from_uris(item, open_func, platform='', only_first=False)]
```

Comment:
why the call to StringIO here? isn't open_func taking care of that?

Reply:
That was the only way of distinguishing between a filename `str` and data already read from the XML file as `str` in `_get_uris_and_open_func()`. Suggestions for a better way are wellcome.